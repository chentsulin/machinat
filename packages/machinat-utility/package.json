{
  "name": "machinat-utility",
  "version": "0.0.1-4",
  "author": "Patrick Liu <zeldalink0515@gmail.com>  (https://github.com/lrills)",
  "repository": {
    "type": "git",
    "url": "https://github.com/lrills/machinat.git",
    "directory": "packages/machinat-utility"
  },
  "homepage": "https://github.com/lrills/machinat#readme",
  "bugs": "https://github.com/lrills/machinat",
  "main": "lib/index.js",
  "files": [
    "lib/*",
    "README.md"
  ],
  "license": "MIT",
  "dependencies": {
    "invariant": "^2.2.4"
  },
  "peerDependencies": {
    "machinat": "^0.0.1-1"
  }
}
