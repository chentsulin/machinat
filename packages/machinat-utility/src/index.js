// @flow
export { default as compose } from './compose';
export { default as isValidElementType } from './isValidElementType';
export { default as formatNode } from './formatNode';
export { default as joinTextualSegments } from './joinTextualSegments';
export { default as valuesOfAssertedType } from './valuesOfAssertedType';
export { default as mixin } from './mixin';
export { default as toJSONWithProto } from './toJSONWithProto';

export * from './isX';
export * from './children';
