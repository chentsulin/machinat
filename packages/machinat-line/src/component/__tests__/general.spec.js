/* eslint-disable no-await-in-loop */
import { map } from 'machinat-utility';
import Machinat from 'machinat';
import generalDelegate from '../general';

const render = async children => {
  const renderings = map(
    children,
    (node, path) =>
      typeof node === 'string'
        ? { type: 'text', value: node, node, path }
        : typeof node.type === 'string'
        ? generalDelegate(node, render, path)
        : { type: 'part', value: { something: 'unknown' }, node, path },
    '$'
  );

  return renderings ? [].concat(...(await Promise.all(renderings))) : null;
};

describe('text components', () => {
  test('shallow textual elements match snapshot', async () => {
    const promise = render([
      <text>abc</text>,
      <a href="https://machinat.world">hello</a>,
      <b>important</b>,
      <i>italic</i>,
      <del>nooooo</del>,
      <br />,
      <code>foo.bar()</code>,
      <pre>foo.bar('hello world')</pre>,
    ]);

    await expect(promise).resolves.toMatchSnapshot();

    const segments = await promise;

    expect(segments.map(s => s.value)).toMatchInlineSnapshot(`
Array [
  "abc",
  "hello",
  Symbol(machinat.segment.break),
  "https://machinat.world",
  Symbol(machinat.segment.break),
  "important",
  "italic",
  "nooooo",
  Symbol(machinat.segment.break),
  "foo.bar()",
  "foo.bar('hello world')",
]
`);
  });

  test('nested textual elements match snapshot', async () => {
    const promise = render(
      <text>
        123{' '}
        <code>
          Hello, <b>Luke Skywalker!</b>
        </code>
        <br />
        You know what?
        <br />
        <i>
          I'm your <del>FATHER</del> <code>droid</code>.
        </i>
        <br />
        <br />
        <a href="https://C3.PO">Check here</a>
        <pre>May the force be with you!</pre> abc
      </text>
    );

    await expect(promise).resolves.toMatchSnapshot();
    const segments = await promise;

    expect(segments.map(r => r.value)).toMatchInlineSnapshot(`
Array [
  "123 Hello, Luke Skywalker!",
  Symbol(machinat.segment.break),
  "You know what?",
  Symbol(machinat.segment.break),
  "I'm your FATHER droid.",
  Symbol(machinat.segment.break),
  Symbol(machinat.segment.break),
  "Check here",
  Symbol(machinat.segment.break),
  "https://C3.PO",
  Symbol(machinat.segment.break),
  "May the force be with you! abc",
]
`);
  });

  test('with break placed in children', async () => {
    const children = (
      <>
        foo
        <br />
        bar
      </>
    );

    const promise = render([
      <text>{children}</text>,
      <a href="https://machinat.world">{children}</a>,
      <b>{children}</b>,
      <i>{children}</i>,
      <del>{children}</del>,
      <code>{children}</code>,
      <pre>{children}</pre>,
    ]);

    await expect(promise).resolves.toMatchSnapshot();
    const segments = await promise;

    expect(segments.map(r => r.value)).toMatchInlineSnapshot(`
Array [
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  Symbol(machinat.segment.break),
  "https://machinat.world",
  Symbol(machinat.segment.break),
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  "foo",
  Symbol(machinat.segment.break),
  "bar",
  "foo",
  Symbol(machinat.segment.break),
  "bar",
]
`);
  });

  test('should throw if non string value rendered', async () => {
    const NonText = () => {};
    const children = (
      <>
        foo
        <NonText />
        bar
      </>
    );

    const elements = [
      <a src="...">{children}</a>,
      <b>{children}</b>,
      <i>{children}</i>,
      <del>{children}</del>,
      <text>{children}</text>,
      <code>{children}</code>,
      <pre>{children}</pre>,
    ];

    for (const element of elements) {
      await expect(render(element)).rejects.toThrow(
        '<NonText /> at $::1 is not valid textual content'
      );
    }
  });

  test('should return null if content is empty', async () => {
    const elements = [
      <a src="..." />,
      <b />,
      <i />,
      <del />,
      <text />,
      <code />,
      <pre />,
    ];

    for (const element of elements) {
      await expect(render(element)).resolves.toEqual([null]);
    }
  });
});

describe('media components', () => {
  it('metch snapshot', async () => {
    const medias = [
      <img src="http://..." />,
      <video src="http://..." />,
      <audio src="http://..." />,
      <file src="http://..." />,
    ];

    for (const media of medias) {
      await expect(render(media)).resolves.toEqual([
        {
          type: 'unit',
          node: media,
          value: {
            text: 'http://...',
            type: 'text',
          },
          path: '$',
        },
      ]);
    }
  });

  test('should return "" if content is empty', async () => {
    const medias = [<img />, <video />, <audio />, <file />];

    for (const media of medias) {
      await expect(render(media)).resolves.toEqual([
        {
          type: 'unit',
          node: media,
          value: { text: '', type: 'text' },
          path: '$',
        },
      ]);
    }
  });
});
