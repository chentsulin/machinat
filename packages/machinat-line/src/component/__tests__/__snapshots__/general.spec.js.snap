// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text components nested textual elements match snapshot 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "123 Hello, Luke Skywalker!",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$:3",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "You know what?",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$:5",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "I'm your FATHER droid.",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$:7",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$:8",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "Check here",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "Check here",
        "href": "https://C3.PO",
      },
      "type": "a",
    },
    "path": "$:9",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "https://C3.PO",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "Check here",
        "href": "https://C3.PO",
      },
      "type": "a",
    },
    "path": "$:9",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Array [
          "123",
          " ",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "Hello, ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "Luke Skywalker!",
                  },
                  "type": "b",
                },
              ],
            },
            "type": "code",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          "You know what?",
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": Array [
                "I'm your ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "FATHER",
                  },
                  "type": "del",
                },
                " ",
                Object {
                  "$$typeof": Symbol(machinat.element),
                  "props": Object {
                    "children": "droid",
                  },
                  "type": "code",
                },
                ".",
              ],
            },
            "type": "i",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {},
            "type": "br",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "Check here",
              "href": "https://C3.PO",
            },
            "type": "a",
          },
          Object {
            "$$typeof": Symbol(machinat.element),
            "props": Object {
              "children": "May the force be with you!",
            },
            "type": "pre",
          },
          " abc",
        ],
      },
      "type": "text",
    },
    "path": "$",
    "type": "text",
    "value": "May the force be with you! abc",
  },
]
`;

exports[`text components shallow textual elements match snapshot 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "abc",
      },
      "type": "text",
    },
    "path": "$:0",
    "type": "text",
    "value": "abc",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "hello",
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "text",
    "value": "hello",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "hello",
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "hello",
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "text",
    "value": "https://machinat.world",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "hello",
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "important",
      },
      "type": "b",
    },
    "path": "$:2",
    "type": "text",
    "value": "important",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "italic",
      },
      "type": "i",
    },
    "path": "$:3",
    "type": "text",
    "value": "italic",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "nooooo",
      },
      "type": "del",
    },
    "path": "$:4",
    "type": "text",
    "value": "nooooo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$:5",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "foo.bar()",
      },
      "type": "code",
    },
    "path": "$:6",
    "type": "text",
    "value": "foo.bar()",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": "foo.bar('hello world')",
      },
      "type": "pre",
    },
    "path": "$:7",
    "type": "text",
    "value": "foo.bar('hello world')",
  },
]
`;

exports[`text components with break placed in children 1`] = `
Array [
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "text",
    },
    "path": "$:0",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "text",
    },
    "path": "$:0",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "text",
    "value": "https://machinat.world",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
        "href": "https://machinat.world",
      },
      "type": "a",
    },
    "path": "$:1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "b",
    },
    "path": "$:2",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "b",
    },
    "path": "$:2",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "i",
    },
    "path": "$:3",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "i",
    },
    "path": "$:3",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "del",
    },
    "path": "$:4",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "del",
    },
    "path": "$:4",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "code",
    },
    "path": "$:5",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "code",
    },
    "path": "$:5",
    "type": "text",
    "value": "bar",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "pre",
    },
    "path": "$:6",
    "type": "text",
    "value": "foo",
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {},
      "type": "br",
    },
    "path": "$::1",
    "type": "break",
    "value": Symbol(machinat.segment.break),
  },
  Object {
    "node": Object {
      "$$typeof": Symbol(machinat.element),
      "props": Object {
        "children": Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "children": Array [
              "foo",
              Object {
                "$$typeof": Symbol(machinat.element),
                "props": Object {},
                "type": "br",
              },
              "bar",
            ],
          },
          "type": Symbol(machinat.element.fragment),
        },
      },
      "type": "pre",
    },
    "path": "$:6",
    "type": "text",
    "value": "bar",
  },
]
`;
